scripts/fuzz_opt.py

```diff
@@ -164,23 +164,64 @@ def randomize_fuzz_settings():
-    os.path.join('lit', 'passes', 'optimize-instructions-gc.wast'),
-    os.path.join('lit', 'passes', 'optimize-instructions-gc-iit.wast'),
-    os.path.join('lit', 'passes', 'optimize-instructions-call_ref.wast'),
-    os.path.join('lit', 'passes', 'inlining_splitting.wast'),
-]
-IMPORTANT_INITIAL_CONTENTS = [os.path.join(shared.get_test_dir('.'), t) for t in IMPORTANT_INITIAL_CONTENTS]
+def get_important_initial_contents():
+    FIXED_IMPORTANT_INITIAL_CONTENTS = [
+        # Perenially-important passes
+        os.path.join('lit', 'passes', 'optimize-instructions.wast'),
+        os.path.join('passes', 'optimize-instructions_fuzz-exec.wast'),
+    ]
+    MANUAL_RECENT_INITIAL_CONTENTS = [
+        # Recently-added or modified passes. These can be added to and pruned
+        # frequently.
+        os.path.join('lit', 'passes', 'once-reduction.wast'),
+        os.path.join('passes', 'remove-unused-brs_enable-multivalue.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-bulk-memory.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-ignore-traps.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-gc.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-gc-iit.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-call_ref.wast'),
+        os.path.join('lit', 'passes', 'inlining_splitting.wast'),
+    ]
+    RECENT_DAYS = 30
+
+    def auto_select_recent_initial_contents():
+        # Returns the list of test wast/wat files added or modified within the
+        # RECENT_DAYS number of days
+        # 1. Print 'git log' with changed file status and without commit message,
+        #    with commits within RECENT_DAYS number of days
+        # 2. Pick up lines in the form of
+        #    A       test/../something.wast
+        #    M       test/../something.wast
+        #    (wat extension is also included)
+        p = re.compile(r'^[AM]\stest' + os.sep + r'(.*\.(wat|wast))$')
+        log = run(['git', 'log', '--name-status', '--format=', '--date=relative', '--no-renames', f'--since="{RECENT_DAYS} days ago"'], silent=True).splitlines()
+        entry_matches = [p.match(e) for e in log]
+        return [match.group(1) for match in entry_matches if match]
+
+    print('- Perenially-important initial contents:')
+    for test in FIXED_IMPORTANT_INITIAL_CONTENTS:
+        print('  ' + test)
+    print()
+
+    recent_contents = []
+    print('- Recently added or modified initial contents ', end='')
+    if shared.options.auto_initial_contents:
+        print(f'(automatically selected: within last {RECENT_DAYS} days):')
+        recent_contents += auto_select_recent_initial_contents()
+    else:
+        print('(manually selected):')
+        recent_contents = MANUAL_RECENT_INITIAL_CONTENTS
+    for test in recent_contents:
+        print('  ' + test)
+    print()
+    if input('Do you want to proceed with these initial contents? (y/n) ').lower() != 'y':
```

Comment:
How about only adding this query when the user changed `shared.options.auto_initial_contents`? Otherwise it would make the default way to run the fuzzer slower.

Reply:
> when the user changed `shared.options.auto_initial_contents`

You mean the user set `shared.options.auto_initial_contents`?

I'm.. little unsure, because I hope this can be the default in the long run. I made it not the default in case it disrupts the recent workflow on GC, but I think, making this automatic is better than expecting every PR submitter to include their test case in this fuzz_opt.py or code reviewer to suggest them to include it, so I didn't want to treat the one more special... But if you think it's inconvenient I can remove it.

And even now, if we set `RECENT_DAYS` to 14 or something, the list is not too different to your manual recent contents. Should we reduce the default to 14?
```
aheejin@aheejin:~/binaryen$ ./scripts/fuzz_opt.py 
- Perenially-important initial contents:
  lit/passes/optimize-instructions.wast
  passes/optimize-instructions_fuzz-exec.wast

- Recently added or modified initial contents (manually selected):
  lit/passes/once-reduction.wast
  passes/remove-unused-brs_enable-multivalue.wast
  lit/passes/optimize-instructions-bulk-memory.wast
  lit/passes/optimize-instructions-ignore-traps.wast
  lit/passes/optimize-instructions-gc.wast
  lit/passes/optimize-instructions-gc-iit.wast
  lit/passes/optimize-instructions-call_ref.wast
  lit/passes/inlining_splitting.wast

Do you want to proceed with these initial contents? (y/n) n
```
```
aheejin@aheejin:~/binaryen$ ./scripts/fuzz_opt.py --auto-initial-contents
- Perenially-important initial contents:
  lit/passes/optimize-instructions.wast
  passes/optimize-instructions_fuzz-exec.wast

- Recently added or modified initial contents (automatically selected: within last 14 days):
  lit/passes/inlining_splitting.wast
  lit/passes/Oz.wast
  lit/wasm-split/imported-memory.wast
  lit/passes/merge-blocks.wast
  heap-types.wast
  lit/passes/optimize-instructions-bulk-memory.wast
  lit/passes/vacuum-intrinsics.wast
  lit/passes/optimize-instructions.wast
  lit/passes/intrinsic-lowering.wast
  lit/passes/optimize-instructions-gc.wast
  lit/passes/optimize-instructions-call_ref.wast
  lit/passes/optimize-instructions-gc-iit.wast
  lit/passes/optimize-instructions-gc-tnh-nn.wast
  lit/passes/inlining-optimizing_optimize-level=3.wast
  passes/Oz_fuzz-exec_all-features.wast
  lit/passes/optimize-instructions-ignore-traps.wast
  passes/remove-unused-brs_enable-multivalue.wast

Do you want to proceed with these initial contents? (y/n) n
```

Oh and one thing; I forgot to `uniq` the auto-computed recent list before, so there were duplicates and the list was longer.

Reply:
Or, I can make this `(Y/n)`, meaning if you just enter, it will be considered as you answered `y`.

Reply:
Made `y` the default option.

Reply:
I think back when we discussed the idea either you or @tlively mentioned that reproducibility would be hurt by enabling this by default. Would we ever be able to enable this by default, that is, can we make this deterministic? The specific problem I worry about is if we look at OS times for files, can we rely on that to be identical between systems? Like if the time resolution is different, maybe the choices could be different, which would be hard to debug. Unless we are certain of determinism, I think this would be useful but should be opt-in.

Reply:
Per @tlively's suggestion, I changed the script to gather files added/modified within the fixed number of days _counting from `HEAD`_ instead of counting from the current system time: https://github.com/WebAssembly/binaryen/pull/4173/commits/a2689eae17a12747660bfefd7e64117241a56cf4

After this I think this will be deterministic given the Binaryen version (commit SHA). I'm not suggesting to change this to default right now though.

Reply:
Oh, nice idea!